scripts/deploy-latest.sh

```diff
@@ -0,0 +1,156 @@
+	local go="${1}"
+
+	local source_dir=$(pwd)
+	echo "START: metakgp-wiki deploy"
+
+	local config_path=${METAKGP_WIKI_PATH:-/root/metakgp-wiki}
+	cd "$config_path" || return 2
+
+	echo "STEP: Ensure that current branch is $base_branch"
+	local branch=$(git rev-parse --abbrev-ref $base_branch)
+	if [[ "$branch" != "$base_branch" ]];
+	then
+		echo "Current branch is not $base_branch. Continue with deployment? (y/N)"
+		read p
+
+		if [[ "$p" != "y" ]];
+		then
+			echo "END: Stopping deployment"
+			return 0
+		fi
+	fi
+
+	echo "STEP: Update repository"
+	git remote update
+
+	echo "STEP: Check if there are any changes that need to be deployed"
+	git --no-pager diff --exit-code $deploy_branch > /dev/null
+
+	if [[ "$?" == "0" ]];
+	then
+		echo "$base_branch and $deploy_branch are the same. Continue with deployment? (y/N)"
+		read p
+
+		if [[ "$p" != "y" ]];
+		then
+			echo "END: Stopping deployment"
+			return 0
+		fi
+	fi
+
+	local docker_compose="docker-compose"
+	local docker="docker"
+
+	# TODO: Fail if docker or docker-compose not found
+	${docker} version
+	${docker_compose} version
+
+	echo "STEP: Running backup job"
+	local backup_container_exec="${docker_compose} -f docker-compose.yml \
+					  -f docker-compose.override.yml \
+					  -f docker-compose.prod.yml exec backup"
+
+	${backup_container_exec} ./run_backup.sh 2>/dev/null
+
+	echo "STEP: Copy backup and store inside $source_dir/.backups"
+
+	mkdir -p "$source_dir/.backups"
+	# TODO: Strange head commands to get rid of the newlines at the end. Use gawk instead?
+	backup_archive="$(${backup_container_exec} ls -1 -t /root/backups 2>/dev/null | head -1 | head -c -2)"
+	backup_container_name=$(${docker} ps  --format '{{ .Names }}' | grep backup | head -1)
+	backup_path="$backup_container_name:/root/backups/$backup_archive"
+	docker cp "$backup_path" "$source_dir/.backups"
+
+	echo "STEP: Deployed version"
+	git log --oneline | head -n1
+
+	if [[ "$go" != "--go" ]];
+	then
+		echo "DRY RUN: Stopping deploy; Run with --go to continue beyond this point"
+		return 0
+	fi
+
+	echo "STEP: Merge branch and build Docker images"
+	git merge --ff-only $deploy_branch
+
+	if [[ "$?" != "0" ]];
+	then
+		echo "ERROR: HEAD does not seem to contain $deploy_branch, can deploy only ff-only branches"
+		return 1
+	fi
+
+	local docker_compose_override="${docker_compose} -f docker-compose.yml \
+					  -f docker-compose.override.yml \
+					  -f docker-compose.prod.yml"
+
+	# TODO: Move to a function
+	if [[ -n "$SLACK_NOTIFICATIONS_URL" ]];
+	then
+		curl -s -H 'content-type: application/json' \
+			 -d "{ \"text\": \"$(deploy_message start)\" }" \
+			 "$SLACK_NOTIFICATIONS_URL"
+	fi
+
+	echo "STEP: Bring all containers down (Downtime begin) $(date +%s)"
+
+	${docker_compose_override} down
+
+	local volume_metakgp="metakgp-wiki_mediawiki-volume"
+	docker volume rm "$volume_metakgp"
+
+	${docker_compose_override} up --build -d
+
+	echo "STEP: Bring all containers down (Downtime end) $(date +%s)"
```

Comment:
Might be clearer to say `STEP: All containers are up (Downtime end)`

For good measure let's also print out `Don't forget to try logging in and editing a page to make sure the wiki is working`?

Reply:
Yeah! That was a typo :white_check_mark: